components/remerge/src/storage/bootstrap.rs

```diff
@@ -36,8 +36,27 @@ pub(super) fn load_or_bootstrap(
 
         if native_ver != native.version.to_string() {
             // XXX migrate existing records here!
-            // XXX Ensure we only move native version forward and not backwards!
+            let native_ver = semver::Version::parse(&*native_ver)
+                .expect("previously-written version is no longer semver");
+            if native.version < native_ver {
+                throw!(ErrorKind::SchemaVersionWentBackwards(
+                    native.version.to_string(),
+                    native_ver.to_string()
+                ));
+            }
             meta::put(db, meta::NATIVE_SCHEMA_VERSION, &native.version.to_string())?;
+        } else {
+            let previous_native: String = db.query_row(
+                "SELECT schema_text FROM remerge_schemas WHERE version = ?",
+                rusqlite::params![native_ver],
+                |r| r.get(0),
+            )?;
+            let previous_native = crate::schema::parse_from_string(&*previous_native, false)?;
```

Comment:
This feels costly but seemed more correct so as not to be confused by whitespace changes.

Reply:
Only happens at startup so I'm not worried.